src/gui/sdl_mapper.cpp

```diff
@@ -2577,19 +2591,10 @@ void MAPPER_Init(void) {
 	}
 }
 
-static void ReloadMapper(Section *sec) {
+void MAPPER_StartUp(Section * sec) {
+	// Set the user-defined mapper file, or the default (if not specified)
 	Section_prop const *const section=static_cast<Section_prop *>(sec);
 	Prop_path const *const pp = section->Get_path("mapperfile");
 	mapper.filename = pp->realpath;
-	GFX_LosingFocus(); //Release any keys pressed, or else they'll get stuck.
-	MAPPER_Init();
-}
-
-void MAPPER_StartUp(Section * sec) {
-	Section_prop * section=static_cast<Section_prop *>(sec);
-	section->AddInitFunction(&ReloadMapper, true); //runs immediately after this function ends
-	mapper.sticks.num=0;
-	mapper.sticks.num_groups=0;
-	memset(&virtual_joysticks,0,sizeof(virtual_joysticks));
 	MAPPER_AddHandler(&MAPPER_Run,MK_f1,MMOD1,"mapper","Mapper");
 }
```

Comment:
`ReloadMapper()` was previously used to inject user's customized mapper as a first step before the mapper is parsed and then initialize the mapper.  However, we can move the file customization directly into the the startup function, and avoid the callback.

We knock out `MAPPER_Init()` from here because config-file parse is done very early in the startup process; even before SDL starts.  So we continue using SDLmain.cpp to run `MAPPER_Init()` after much of SDL has been initialized; enough to get valid joystick data back from it.